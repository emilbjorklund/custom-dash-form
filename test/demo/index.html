<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Expected outcome of form validaton</title>
  <script defer nomodule src="../../dist/demo.js"></script>
  <script defer type="module" src="../../src/demo.es6.js"></script>

  <style>
    *,
    *:before,
    *:after {
      box-sizing: border-box;
    }
    custom-form,
    custom-formcontrol {
      display: block;
    }

    body {
      font-family: Avenir Next, Avenir, SegoeUI, arial, sans-serif;
      line-height: 1.5;
    }
    .form-control {
      padding: 1em 0;
      max-width: 35em;
      margin: 1.25rem 0;
    }
    
    .form-control + .form-control {
      border-top: 1px solid;
    }

    .error {
      outline: 1px solid red;
    }

    label {
      display: block;
    }

    .error-label {
      color: red;
    }

    input:not([type="radio"]):not([type="checkbox"]) {
      width: 100%;
      padding: .25em;
      font: inherit;
    }

    .error-message {
      border: 1px solid red;
      color: red;
      margin: 1em 0;
      padding: .5em;
    }
  </style>
</head>
<body>
  
  <h1>Form validation demo!</h1>

  <custom-form>
    <form action=".">
      <div class="form-control">
        <label for="non-custom-field">Non custom required text input</label>
        <input type="text" id="non-custom-field" required>
      </div>

      <div class="form-control">
        <label for="non-custom-select">Select something here, please:</label>
        <select name="non-custom-select" required id="non-custom-select">
          <option value="">(No selection made)</option>
          <option value="1">Option 1</option>
          <option value="2">Option 2</option>
        </select>
      </div>

      <div class="form-control">
        <label for="non-custom-select-optional">This one is optional:</label>
        <select name="non-custom-select-optional" id="non-custom-select-optional">
          <option value="">(No selection made)</option>
          <option value="1">Option 1</option>
          <option value="2">Option 2</option>
        </select>
      </div>
      
      <div class="form-control">
        <label>
          <input id="non-custom-checkbox" type="checkbox" required>
          Please check this box
        </label>
      </div>

      <custom-formcontrol class="form-control" valuemissing="Du måste verkligen fylla i den här.">
        <label>
          <input id="custom-checkbox" type="checkbox" required>
          Please check this box, too.
        </label>
      </custom-formcontrol>

      <custom-formcontrol class="form-control" valuemissing="Hallå, du måste skriva nåt här." tooshort="Alltså, typ minst 20 tecken." typemismatch="Ser inte ut att vara en giltig URL...">
        <label for="custom-field">Custom required URL input</label>
        <input type="url" minlength="20" id="custom-field" required>
      </custom-formcontrol>

      <custom-formcontrol class="form-control" valuemissing="Hallå, du måste skriva en siffra." stepmismatch="Måste vara ett jämnt tal!">
        <label for="custom-field-number">Custom required number input</label>
        <input type="number" required min="0" max="10" step="2" value="3" id="custom-field-number">
      </custom-formcontrol>

      <button type="submit">Send</button>
      <button type="submit" formnovalidate>Send without validation</button>
    </form>
  </custom-form>
</body>
</html>