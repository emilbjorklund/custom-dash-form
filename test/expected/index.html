<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Expected outcome of form validaton</title>
  <script defer src="../../src/customForm.js"></script>
</head>
<body>
  
  <h1>Expected outcome of form validation</h1>
  
  <h2>Before</h2>
  <custom-form>
    <form action=".">
      <div>
        <label for="non-custom-field">Non custom required text input</label>
        <input type="text" id="non-custom-field" required>
      </div>
      <custom-formcontrol valuemissing="Du måste skriva nåt här." typemismatch="Ser inte ut att vara en giltig epostadress...">
        <label for="custom-field">Custom required text input</label>
        <input type="text" minlength="2" id="custom-field" required>
      </custom-formcontrol>

      <button type="submit">Send</button>
    </form>
  </custom-form>

  <h2>After</h2>
  <custom-form>
    <form action="/foo" novalidate>
      <div>
        <label for="non-custom-field">Non custom required text input</label>
        <input type="text" required>
        <div class="custom-form-error">
          <p>(whatever the browser uses for error message)</p>
        </div>
      </div>
      <custom-formcontrol valuemissing="Du måste skriva nåt här.">
        <label for="custom-field">Custom required text input</label>
        <input type="text" required>
        <div class="custom-form-error">
          <p>Du måste skriva nåt här</p>
        </div>
      </custom-formcontrol>

      <button type="submit">Send</button>
    </form>
  </custom-form>
</body>
</html>